adr/minimal-sdk.md

```diff
@@ -0,0 +1,53 @@
+3. **New, Full-Featured Red Hat Container/Service:** The same as above, but delivered as a containerized HTTP service. This could simplify agent sharing and support a broader assortment of programming languages, but it adds operational complexity for developers.
+4. **Adopt a Single Third-Party SDK:** Formally adopt, package, and recommend a single existing framework (e.g., LangChain) as the "official" way to build agents on our platform. This would leverage a mature ecosystem but cedes strategic control and risks lock-in to that framework's architecture and release cadence.
+5. **Support Multiple Co-Equal SDKs:** Officially support and document several third-party SDKs. This offers maximum choice but makes it difficult for our own platform tools (e.g., a UI for creating agents) to provide a consistent, functional user experience, as they would need to target multiple, incompatible outputs.
+6. **New, Minimal Red Hat Container/Service:** Provide an ultra-lightweight container that exposes a simple HTTP API for agent interaction and recommend users graduate to more powerful frameworks for complex needs.
+7. **New, Minimal Red Hat SDK:** Provide an ultra-lightweight SDK designed for basic use cases and tutorials, while officially recommending that customers adopt comprehensive third-party frameworks for advanced applications.
+
+## Decision
+
+We will adopt **Option 7**. We will create a **new, minimal, lightweight Red Hat Agent SDK** and concurrently establish a strategy to **officially recommend and ensure compatibility with established third-party frameworks**.
+
+The new Red Hat Agent SDK will be written in **Python**. The specific list of third-party frameworks we will recommend and ensure compatibility with will be addressed in one or more future ADRs.
+
+The key characteristics of the Red Hat Agent SDK will be:
+
+* **Minimalist Scope:** It will focus exclusively on implementing the core agent loop (state management, API calls, tool dispatching) as a reference. It is not intended to have complex features like chains, memory modules, or vector store integrations.
+* **Enablement, Not Competition:** Its primary purpose is to serve as a clear, simple "golden path" for new users, tutorials, documentation, and basic proof-of-concept applications.
+* **Reference Implementation:** It will act as the reference for how to correctly interact with the Llama Stack Responses API in an agentic fashion, providing a foundation that other tools can build upon or refer to.
+
+This two-pronged approach (a minimal "first-step" SDK and a clear "next-step" recommendation of powerful external tools) provides the best balance of speed, user choice, and focused engineering effort.
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Positive
+
+* **Rapid Time-to-Market:** A minimal SDK is a fast way to deliver a functional agent construct to our users, allowing us to provide a complete end-to-end story quickly.
```

Comment:
How shallow will this be? How different will that be compared to the existing Agent implementation in llama stack (which needs work and is still minimal)?

e.g., I want to build an AI system that can diagnose my cluster, provide recommendations on next steps, fixes, do the fixes, or file issues, ...

Reply:
> How shallow will this be?

I think just deep enough to cover our onboarding experience and the things you can do in our agent playground.  Since we control it, we can adjust it as needed as the agent playground design evolves.

> How different will that be compared to the existing Agent implementation in llama stack (which needs work and is still minimal)?

The main differences are:

1. It will still work if/when the Agent API in Llama Stack undergoes breaking changes and/or are deprecated + removed.
2. It runs on the client, not the server.

The motivation for doing it is the former, not the latter.

Reply:
I think the client-server distinction is key.

Reply:
What if we had the playground UI use the Responses API and we'll see if the need for a specialized client-side library grows out of that or not?